src/Integration.Vsix/CFamily/CFamilyHelper.cs

```diff
@@ -102,14 +102,20 @@ private static Request CreateRequest(FileConfig fileConfig, string absoluteFileP
                 }
             }
 
+            ICFamilyRulesConfig rulesConfig = null;
             if (!isPCHBuild)
             {
                 // We don't need to calculate / set the rules configuration for PCH builds
-                request.RulesConfiguration = cFamilyRulesConfigProvider.GetRulesConfiguration(request.CFamilyLanguage);
-                Debug.Assert(request.RulesConfiguration != null, "RulesConfiguration should be set for the analysis request");
-                request.Options = GetKeyValueOptionsList(request.RulesConfiguration);
+                rulesConfig = cFamilyRulesConfigProvider.GetRulesConfiguration(request.CFamilyLanguage);
+                Debug.Assert(rulesConfig != null, "RulesConfiguration should be have been retrieved");
+                request.Options = GetKeyValueOptionsList(rulesConfig);
             }
 
+            var context = new RequestContext(request.CFamilyLanguage, rulesConfig, request.File, SubProcessFilePaths.PchFilePath,
+                analyzerOptions as CFamilyAnalyzerOptions);
+
+            request.Context = context;
```

Comment:
For now, the `CFamilyHelper` just creates a new context object at that duplicates some of the information in the `Request/FileConfig`.

This would be cleaned up as part of moving to the factory model.